teleduck/tests/test_duckdb_catalog.py

```
@@ -49,10 +50,20 @@ def setUpClass(cls):
             cls.proc.join()
             raise RuntimeError("Server did not start")
 
+        # the socket binds before the pg_catalog emulation is query-ready, so
+        # poll the exact entry the tests assert on (the registered database in
+        # pg_database) until it answers instead of sleeping a fixed amount.
+        waited = wait_for_catalog(
+            cls.port,
+            "db",
+            f"SELECT datname FROM pg_catalog.pg_database WHERE datname='{cls.database_name}'",
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)